syft/frameworks/torch/nn/functional.py

```diff
@@ -150,3 +150,56 @@ def conv2d(input, weight, bias=None, stride=1, padding=0, dilation=1, groups=1):
         .contiguous()
     )
     return res
+
+
+def _pool(tensor, kernel_size: int = 2, stride: int = 2, mode="max"):
+    output_shape = (
+        (tensor.shape[0] - kernel_size) // stride + 1,
+        (tensor.shape[1] - kernel_size) // stride + 1,
+    )
+    kernel_size = (kernel_size, kernel_size)
+    b = torch.ones(tensor.shape)  # when torch.Tensor.stride() is supported: replace with A.stride()
+    a_strides = b.stride()
+    a_w = torch.as_strided(
+        tensor,
+        size=output_shape + kernel_size,
+        stride=(stride * a_strides[0], stride * a_strides[1]) + a_strides,
+    )
+    a_w = a_w.reshape(-1, *kernel_size)
+    result = []
+    if mode is "max":
+        for channel in range(a_w.shape[0]):
+            result.append(a_w[channel].max())
+    else:
```

Comment:
It be good if the `else` captures just the `mean` arg with other modes leading to raising Exceptions

Reply:
well spotted! I actually thought about that but I made it that way to avoid writing a test case for the expectation 😂 will do it next commit